### ISSUE TITLE
Simulate greater batches

### ISSUE THREAD

**Issue by petrbel** [Cognexa/cxflow-tensorflow#19]
@FloopCZ :

> Yes, we will have to slightly hack our own framework. :-)

> Basically we would override run() method of the BaseTF net to only update the weights every N'th batch. During the other N batches, it would only aggregate the [gradients](https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer#compute_gradients) and every N'th batch it would [apply](https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer#apply_gradients) the mean of the gradients. 

...

> I think it should be that straightforward. The new gradient-accumulating run() function in OurNet replaces the BaseTFNet's run() function, which will not be used at all, thus we don't have to pass it anything. The only thing that has to be taken care of is properly averaging the accumulated gradients by a weighted average with respect to batch sizes.
> 
> And yes, maybe our BaseTFNet could support larger batch emulation by itself.

**Comment by blazekadam** [comment]
With multi-GPU support, layer norm etc., this feature seems to be unimportant.

**Comment by petrbel** [comment]
Does it? In some cases, with N GPUs we'll have only batch-size=N which I expect to be 2-8. Not that much...

**Comment by blazekadam** [comment]
I have never seen this feature in action. Among all *may be useful* sort of features this one is not among the tops, I see no point of having this issue open.

**Comment by petrbel** [comment]
after discussion with @blazekadam I'm reopening this issue and labeling it to low prio
